x-pack/platform/plugins/shared/onechat/public/application/context/send_message/use_subscribe_to_chat_events.ts

```diff
@@ -68,6 +71,33 @@ export const useSubscribeToChatEvents = ({
                 tool_id: event.data.tool_id,
               }),
             });
+
+            // Check if this is a browser tool call and execute it immediately
+            const toolId = event.data.tool_id;
+            if (toolId && toolId.startsWith('browser_') && browserToolExecutor && browserApiTools) {
+              const originalToolId = toolId.substring('browser_'.length);
+
+              const toolDef = browserApiTools.find((tool) => tool.id === originalToolId);
+              if (toolDef) {
+                const toolsMap = new Map([[originalToolId, toolDef]]);
+                browserToolExecutor
+                  .executeToolCalls(
```

Comment:
I'm wondering if we should maybe have a distinct event type rather than `toolCall` for browser-side tools. I feel like those are different features, so we probably want to keep the events (and round step) dissociated